metis-indexing/src/main/java/eu/europeana/indexing/solr/property/AggregationSolrCreator.java

```diff
@@ -15,38 +26,113 @@
+        intermediatePair.getRight());
+
     SolrPropertyUtils
-        .addValues(doc, EdmLabel.PROVIDER_AGGREGATION_EDM_PROVIDER, aggr.getEdmProvider());
-    SolrPropertyUtils.addValues(doc, EdmLabel.PROVIDER_AGGREGATION_EDM_HASVIEW, aggr.getHasView());
+        .addValues(doc, EdmLabel.PROVIDER_AGGREGATION_DC_RIGHTS, aggregation.getDcRights());
+    if (!SolrPropertyUtils.hasLicenseForRights(aggregation.getEdmRights(),
+        item -> licenses.stream().anyMatch(license -> license.getAbout().equals(item)))) {
+      SolrPropertyUtils
+          .addValues(doc, EdmLabel.PROVIDER_AGGREGATION_EDM_RIGHTS, aggregation.getEdmRights());
+    }
     SolrPropertyUtils
-        .addValue(doc, EdmLabel.PROVIDER_AGGREGATION_EDM_IS_SHOWN_AT, aggr.getEdmIsShownAt());
+        .addValues(doc, EdmLabel.PROVIDER_AGGREGATION_EDM_HASVIEW, aggregation.getHasView());
+    SolrPropertyUtils.addValue(doc, EdmLabel.PROVIDER_AGGREGATION_EDM_IS_SHOWN_AT,
+        aggregation.getEdmIsShownAt());
+    SolrPropertyUtils.addValue(doc, EdmLabel.PROVIDER_AGGREGATION_EDM_IS_SHOWN_BY,
+        aggregation.getEdmIsShownBy());
     SolrPropertyUtils
-        .addValue(doc, EdmLabel.PROVIDER_AGGREGATION_EDM_IS_SHOWN_BY, aggr.getEdmIsShownBy());
-    SolrPropertyUtils.addValue(doc, EdmLabel.PROVIDER_AGGREGATION_EDM_OBJECT, aggr.getEdmObject());
-    SolrPropertyUtils.addValue(doc, EdmLabel.EDM_UGC, aggr.getEdmUgc());
-    doc.addField(EdmLabel.PREVIEW_NO_DISTRIBUTE.toString(), aggr.getEdmPreviewNoDistribute());
-    new WebResourceSolrCreator(licenses).addAllToDocument(doc, aggr.getWebResources());
+        .addValue(doc, EdmLabel.PROVIDER_AGGREGATION_EDM_OBJECT, aggregation.getEdmObject());
+    SolrPropertyUtils.addValue(doc, EdmLabel.EDM_UGC, aggregation.getEdmUgc());
+    doc.addField(EdmLabel.PREVIEW_NO_DISTRIBUTE.toString(),
+        aggregation.getEdmPreviewNoDistribute());
+    new WebResourceSolrCreator(licenses).addAllToDocument(doc, aggregation.getWebResources());
+  }
+
+  private Pair<List<String>, Map<String, List<String>>> splitUrisFromLiterals(
+      final Map<String, List<String>> urisLiteralsMap) {
+    final List<String> uriList = new ArrayList<>();
+    final Map<String, List<String>> literalsMap = new HashMap<>();
+
+    if (MapUtils.isNotEmpty(urisLiteralsMap)) {
+      extractUris(urisLiteralsMap, uriList, literalsMap);
+
+      //Extend map with organization pref labels
+      if (CollectionUtils.isNotEmpty(uriList)) {
+        extendWithOrganizationLiterals(literalsMap);
+      }
+    }
+    return new ImmutablePair<>(uriList, literalsMap);
+  }
+
+  private void extractUris(Map<String, List<String>> urisLiteralsMap,
+      List<String> uriList, Map<String, List<String>> literalsMap) {
+    for (Map.Entry<String, List<String>> entry : urisLiteralsMap.entrySet()) {
+      final List<String> literals = new ArrayList<>();
+      for (String value : entry.getValue()) {
+        if (organizationMap.containsKey(value)) {
+          uriList.add(value);
```

Comment:
So we 'lose' the key (i.e. the language) of these links? There is no danger with this?

Reply:
Correct! But we don't really lose anything, since the links shouldn't have a language in the first place. (well with the conversion they are getting the "def" language which is then cleared if it's empty..
So I don't think there is danger in that perspective.